### PR TITLE
Add working v4.0.2 config to README

### LINKED ISSUE #1: Not recognized by strapi version 4
I can't tell from the README/versioning if this is expected to work with the most recent version of Strapi, but when I've set this up as below, it doesn't seem as though Strapi is even recognizing the plugin:

```js
module.exports = ({ env }) => ({
  // ...
  upload: {
    provider: "aws-s3-advanced",
    providerOptions: {
      accessKeyId: env("STRAPI_AWS_ACCESS_KEY_ID"),
      secretAccessKey: env("STRAPI_AWS_ACCESS_SECRET"),
      region: env("STRAPI_AWS_REGION"),
      params: {
        Bucket: env("STRAPI_AWS_BUCKET"),
      },
      baseUrl: env("CF_BASE_URL"), // e.g. https://cdn.example.com, this is stored in strapi's database to point to the file
      prefix: env("STRAPI_AWS_BUCKET_PREFIX"), // e.g. strapi-assets, note the missing slash at the start
    },
  },
// ...
})
```
I've changed the env variables to ones that do not exist and no errors are thrown, which leads me to believe that the config itself is not being read by strapi. 

When I added a `config: {... }` wrapper around everything within the `upload` block, so that I could put `enabled: true` in the upper block

```js
  upload: {
    enabled: true,
    config: {
      provider: "aws-s3-advanced",
      providerOptions: {
        accessKeyId: env("STRAPI_AWS_ACCESS_KEY_ID"),
        secretAccessKey: env("STRAPI_AWS_ACCESS_SECRET"),
        region: env("STRAPI_AWS_REGION"),
        params: {
          Bucket: env("STRAPI_AWS_BUCKET"),
        },
        baseUrl: env("CF_BASE_URL"), // e.g. https://cdn.example.com, this is stored in strapi's database to point to the file
        prefix: env("STRAPI_AWS_BUCKET_PREFIX"), // e.g. strapi-assets, note the missing slash at the start
      },
    },
  },
```

it did throw the below:

```sh
[2021-12-22 18:07:09.846] error: Could not load upload provider "aws-s3-advanced".
```

### PR DESCRIPTION
**Pull request by aaronclimbs**
This was how I was able to get it working for me - the format of the upload provider config was slightly different and needed to be adapted to version 4. I think this should potentially be above the config for <4.0.0, but I leave that to you. 

This would close #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,15 +23,77 @@ module.exports = ({ env }) => ({
       secretAccessKey: env('AWS_ACCESS_SECRET'),
       region: env('AWS_REGION'),
       params: {
-        Bucket: env('AWS_BUCKET'),
+        bucket: env('AWS_BUCKET'),
```
Comment (zoomoid) [inline README.md:26]: Thanks for including this, I somehow missed this 👍

File: README.md [hunk 2]
```diff
@@ -24,15 +23,77 @@ module.exports = ({ env }) => ({
...
     },
   },
   // ...
 });
 ```
+#### Image Previews
+
+To allow the thumbnails to properly populate, add the below config to
+
+`./config/plugins.js`
```
Comment (zoomoid) [inline README.md:66]:
```suggestion
`./config/middlewares.js`
```

According to <https://github.com/strapi/strapi/tree/master/packages/providers/upload-aws-s3>, this should be in `./config/middlewares.js`, which also resolves my confussion regarding the return type of the function exported from plugins.js

File: README.md [hunk 3]
```diff
@@ -24,15 +23,77 @@ module.exports = ({ env }) => ({
...
+    config: {
+      contentSecurityPolicy: {
+        useDefaults: true,
+        directives: {
+          'connect-src': ["'self'", 'https:'],
+          'img-src': [
+            "'self'",
+            'data:',
+            'blob:',
+            `${env('CF_BASE_URL')}`
```
Comment (zoomoid) [inline README.md:81]:
```suggestion
            `${env('CDN_BASE_URL')}`
```
The config above uses the `CDN_BASE_URL` env variable as reference to a Cloudfront endpoint, I imagine `CF_BASE_URL` is an additional one created by yourself? Could you maybe converge this?

File: README.md [hunk 4]
```diff
@@ -24,15 +23,77 @@ module.exports = ({ env }) => ({
...
+          ],
+          'media-src': [
+            "'self'",
+            'data:',
+            'blob:',
+            `${env('CF_BASE_URL')}`
```
Comment (zoomoid) [inline README.md:87]: same as above 👍

**Review by zoomoid — CHANGES_REQUESTED** [review]
Good job on this 👏

**Comment by zoomoid** [comment]
With the adoptions requested above I was able to reproduce that the plugin works with strapi >= 4.0.0, so this is good to go after the changes 🎉

File: README.md (hunk 4 above)
Comment (aaronclimbs) [reply README.md]: All set!

File: README.md (hunk 3 above)
Comment (aaronclimbs) [reply README.md]: You're right about that - I did leave those in there from my own Cloudfront config. Fixed here.

File: README.md (hunk 2 above)
Comment (aaronclimbs) [reply README.md]: You're absolutely right - this was just an oversight on my part.

**Comment by aaronclimbs** [comment]
These comments should be addressed now

**Review by zoomoid — APPROVED** [review]
Looks great, thank you so much!

**Pull request merged**